### ISSUE TITLE
Warning (emacs): [XELB] Retrieve reply timeout

### ISSUE THREAD

**Issue by thomasheartman** [ch11ng/xelb#27]
Hi,

I recently started getting this warning whenever I start EXWM and I don't know what it means:
```
Warning (emacs): [XELB] Retrieve reply timeout
```

Can you tell me what it means and if it's anything to worry about? I've searched around but can't find any references to it in other issues or with other internet searches.

I haven't noticed anything not working and it doesn't seem to be causing any problems, but I'd rather find out of this now than when it's causing some unexpected issues further down the line. 

Most appreciated. Cheers!

**Comment by ch11ng** [comment]
This basically tells some X request/reply is not processed normally.  Did you made any change to your configuration?

And are you working remotely? If so you may try adjusting `xcb:connection-timeout` to a larger value (the default is 3 seconds which is enough for local use) and see if the phenomenon persists.

**Comment by thomasheartman** [comment]
Yes, I have made some changes to my configuration recently, so that's probably the cause of it. However, the issue seems to have resolved itself now. Or at least: I didn't get the warning on the most recent startup. Is there a way to figure out which request this is about? And why it's failing?

No, I'm not working remotely at the moment, but thanks for the tip! I'll try and keep it in mind if I do some time.

**Comment by ch11ng** [comment]
> Is there a way to figure out which request this is about? And why it's failing?

I guess this happens in connection setup phase and is likely a client-side (EXWM) issue.  If you manage to save a backtrace then it's possible to tell where it fails.  As you were not working remotely it's of course not a networking issue.  Perhaps some Emacs package interferes with EXWM/XELB and you can verify this with a minimal setup.

**Comment by matthuszagh** [comment]
I just wanted to share that I'm experiencing this issue too. I notice it when using Firefox. After a while, firefox becomes extremely sluggish and I start to see this error. I appreciate this comment isn't very helpful for narrowing down the issue, but I still wanted to report it. If I'm able to get a backtrace I'll follow up with it.
